Initial basic support for Whirlpool A/C messages.

crankyoldgit opened the pull request:
- Send Whirlpool A/C messages.
- Basic decoding of Whirlpool A/C messages. Settings are not decoded.
- Add support to IRMQTTServer example code.
- Very basic unit tests for send/decoding.

Ref: #509

crankyoldgit committed:
Initial basic support for Whirlpool A/C messages.
- Send Whirlpool A/C messages.
- Basic decoding of Whirlpool A/C messages. Settings are not decoded.
- Add support to IRMQTTServer example code.
- Very basic unit tests for send/decoding.

Ref: #509

crankyoldgit committed:
Initial basic support for Whirlpool A/C messages.
- Send Whirlpool A/C messages.
- Basic decoding of Whirlpool A/C messages. Settings are not decoded.
- Add support to IRMQTTServer example code.
- Very basic unit tests for send/decoding.

Ref: #509

crankyoldgit commented:
FYI @redmusicxd

crankyoldgit committed:
Refactor decodeWhirlpoolAC() to remove duplicate code.

crankyoldgit committed:
Refactor decodeWhirlpoolAC() to remove duplicate code.

crankyoldgit committed:
Correct a typo in a comment.

crankyoldgit committed:
Correct a typo in a comment.

rednblkx commented:
I'll give it a test when i have time.

crankyoldgit committed:
Better inter-section gap detection for Whirlpool A/C decoding.

crankyoldgit committed:
Better inter-section gap detection for Whirlpool A/C decoding.

rednblkx commented:
I uploaded the code for IRrecvDumpV2 with the modifications (of course 😁) and here is the first data:
Timestamp : 000187.646
Encoding  : WHIRLPOOL_AC
Code      : C160268E0000296C0000000000ED00800000000080 (168 bits)
Library   : v2.4.3

uint16_t rawData[343] = {8944, 4488,  594, 1644,  594, 1648,  600, 526,  592, 540,  600, 532,  598, 540,  600, 540,  598, 1650,  600, 520,  598, 1642,  596, 1648,  600, 530,  600, 534,  596, 542,  598, 544,  596, 534,  594, 526,  592, 530,  598, 1646,  594, 536,  592, 542,  596, 1658,  592, 1666,  592, 538,  592, 1646,  594, 530,  600, 526,  602, 528,  600, 1652,  596, 1658,  602, 1658,  592, 540,  600, 520,  598, 526,  594, 534,  596, 536,  594, 540,  600, 538,  592, 550,  600, 530,  600, 522,  598, 526,  592, 534,  596, 536,  594, 540,  598, 540,  600, 540,  600, 514,  594, 7920,  592, 528,  602, 522,  596, 1648,  592, 538,  600, 1652,  598, 540,  600, 542,  598, 1650,  600, 520,  598, 1644,  594, 1650,  598, 532,  598, 1654,  594, 1662,  598, 542,  598, 534,  596, 524,  596, 528,  600, 528,  602, 528,  600, 534,  594, 542,  598, 544,  596, 534,  596, 526,  594, 530,  600, 528,  590, 540,  600, 534,  596, 542,  598, 542,  596, 534,  596, 526,  594, 530,  600, 528,  592, 540,  600, 534,  596, 540,  598, 544,  598, 532,  596, 524,  594, 530,  600, 528,  602, 530,  600, 534,  596, 542,  598, 544,  594, 536,  604, 518,  602, 522,  596, 532,  598, 532,  596, 538,  592, 546,  594, 548,  592, 540,  600, 1638,  600, 1642,  598, 1648,  592, 538,  600, 1652,  598, 1658,  592, 550,  602, 1630,  598, 7920,  594, 526,  594, 530,  598, 528,  602, 530,  600, 536,  594, 544,  596, 546,  596, 536,  594, 1646,  594, 530,  600, 528,  602, 530,  600, 536,  594, 544,  596, 546,  594, 538,  592, 528,  600, 524,  596, 532,  598, 532,  596, 538,  592, 546,  594, 548,  592, 540,  600, 520,  600, 524,  594, 534,  596, 534,  594, 540,  600, 538,  602, 540,  600, 530,  598, 522,  596, 528,  592, 536,  594, 538,  600, 534,  596, 542,  598, 544,  596, 536,  594, 528,  592, 532,  596, 532,  598, 532,  598, 536,  594, 546,  594, 546,  594, 538,  592, 1646,  592, 532,  598, 528,  600, 530,  600, 534,  594, 544,  596, 546,  594, 518,  590};  // WHIRLPOOL_AC
uint8_t state[21] = {0xC1, 0x60, 0x26, 0x8E, 0x00, 0x00, 0x29, 0x6C, 0x00, 0x00, 0x00, 0x00, 0x00, 0xED, 0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x80};

It does indeed output the whole 168-bit code.

rednblkx commented:
So next i will press different buttons and combinations on the remote & record them && I'll post them in my git repo after i merge this implementation.

crankyoldgit committed:
Bump version of IRMQTTServer after adding basic Whirlpool support.

crankyoldgit committed:
Bump version of IRMQTTServer after adding basic Whirlpool support.

crankyoldgit committed:
Merge branch '2.5-dev-whirlpool_ac' of https://github.com/markszabo/IRremoteESP8266 into 2.5-dev-whirlpool_ac

crankyoldgit committed:
Change bit order of Whirlpool A/C Messages
Analysis of the temperature data in the Whirlpool messages
indicate the bytes of data are in LSB First order, not MSB First.

See https://github.com/markszabo/IRremoteESP8266/issues/509#issuecomment-417918572

crankyoldgit committed:
Add supported model information to Whirlpool

crankyoldgit commented:
No review comments in over a week. Merging.
